react/src/components/search-app/SearchResultsComponent.jsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+
+
+export default class SearchResultsComponent extends React.Component {
```

Comment:
I think it is very suitable, because everywhere where user searches via url he would also be presented a nice search field with his query and could modify it whenever he wishes. But I agree with you that the SearchBarComponent should be split into two - one part for fetching the data and one part to display the results (To separate functional and presentational logic)

The functional component queries and does the heavy lifting and propagates the data to presentational which renders it to achieve separation of concerns, but let's hear the other opinions